packages/backend/src/models/ProjectModel/ProjectModel.ts

```diff
@@ -442,4 +442,29 @@ export class ProjectModel {
             await func();
         });
     }
+
+    async getWarehouseFromCache(
+        projectUuid: string,
+    ): Promise<WarehouseCatalog | undefined> {
+        const warehouses = await this.database(CachedWarehouseTableName)
+            .select(['warehouse'])
+            .where('project_uuid', projectUuid)
+            .limit(1);
+        if (warehouses.length > 0) return warehouses[0].warehouse;
+        return undefined;
+    }
+
+    async saveWarehouseToCache(
+        projectUuid: string,
+        warehouse: WarehouseCatalog,
+    ): Promise<{ projectUuid: string; warehouse: WarehouseCatalog }[]> {
```

Comment:
Did you return to return a list of results? I think you need to select the returning row:

```typescript
const [row] = await this. database(....)
return row;
```